frontend/src/metabase/query_builder/reducers.js

```diff
@@ -560,19 +560,27 @@ export const visibleTimelineIds = handleActions(
   [],
 );
 
-export const hiddenTimelineEventIds = handleActions(
+export const visibleTimelineEventIds = handleActions(
   {
-    [INITIALIZE_QB]: { next: () => [] },
-    [HIDE_TIMELINE_EVENTS]: {
-      next: (state, { payload: events }) => [...state, events.id],
+    [INITIALIZE_QB]: { next: () => null },
+    [SHOW_TIMELINES]: {
+      next: (state, { payload: timelines }) =>
+        // We want to populate all event ids on a first SHOW_TIMELINES.
+        // In subsequent ones, we want to show the previously populated ids.
+        state === null
```

Comment:
I believe it would be more clear to retain the initial state as `[]`, remove this `null` check and call `showTimelineEvents` from there https://github.com/metabase/metabase/blob/1aa9565c451fb2c92c4cab46c45dde7a0adab174/frontend/src/metabase/query_builder/actions/timelines.js#L24
something like 
```
dispatch(showTimelineEvents(collectionTimelines.flatMap(t => t.events));
```